File: 백준/Silver/10845. 큐/큐.py
queue = []
rear = 0
fr = 0

def push(X):
    global rear
    queue.append(X)  # 정수 X를 큐에 넣는 연산
    rear = len(queue)
    return X

def pop():
    global fr
    if rear == fr:  # 큐가 비어 있는 경우
        return -1
    res = fr
    fr += 1
    return queue[res]

def size():
    return rear - fr

def empty():
    if rear == fr:
        return 1
    else:
        return 0

def front():
    if rear == fr:  # 큐가 비어 있는 경우
        return -1
    else:
        return queue[fr]

def back():
    if rear == fr:  # 큐가 비어 있는 경우
        return -1
    else:
        return queue[rear - 1]  # `rear`는 배열의 길이를 나타내므로 마지막 원소는 `rear - 1` 위치

# 명령어 처리 부분
import sys
N = int(sys.stdin.readline().rstrip())
for _ in range(N):
    command = sys.stdin.readline().split()

    if command[0] == 'push':
        push(int(command[1]))
    elif command[0] == 'pop':
        print(pop())
    elif command[0] == 'size':
        print(size())
    elif command[0] == 'empty':
        print(empty())
    elif command[0] == 'front':
        print(front())
    elif command[0] == 'back':
        print(back())

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static int[] queue = new int[10001]; // 큐 배열
    static int first = 0;
    static int last = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push":
                    push(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    sb.append(pop()).append("\n");
                    break;
                case "size":
                    sb.append(size()).append("\n");
                    break;
                case "empty":
                    sb.append(empty()).append("\n");
                    break;
                case "front":
                    sb.append(front()).append("\n");
                    break;
                case "back":
                    sb.append(back()).append("\n");
                    break;
            }
        }
        System.out.print(sb);
    }

    private static int back() {
        if (first == last) {
            return -1;
        }
        return queue[last - 1];
    }

    private static int front() {
        if (first == last) {
            return -1;
        }
        return queue[first];
    }

    private static int empty() {
        return (first == last) ? 1 : 0;
    }

    private static int size() {
        return last - first;
    }

    private static int pop() {
        if (first == last) {
            return -1;
        }
        return queue[first++];
    }

    private static void push(int num) {
        queue[last++] = num;
    }
}